cmd/earthly/subcmd/build_cmd.go

```diff
@@ -579,14 +583,48 @@ func (a *Build) ActionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs []stri
 		return fmt.Errorf("configuration error: \"conversion_parallelism\" must be larger than zero")
 	}
 	parallelism := semutil.NewWeighted(int64(a.cli.Cfg().Global.ConversionParallelism))
+
 	localRegistryAddr := ""
 	if isLocal && a.cli.Flags().LocalRegistryHost != "" {
-		lrURL, err := url.Parse(a.cli.Flags().LocalRegistryHost)
+		u, err := url.Parse(a.cli.Flags().LocalRegistryHost)
 		if err != nil {
 			return errors.Wrapf(err, "parse local registry host %s", a.cli.Flags().LocalRegistryHost)
 		}
-		localRegistryAddr = lrURL.Host
+		localRegistryAddr = u.Host
+	}
+
+	if a.cli.Flags().UseRemoteRegistry {
+		tmpAddr := "localhost:0" // Have the OS select a port
+
+		ln, err := listener.NewListener("tcp", tmpAddr)
+		if err != nil {
+			return errors.Wrapf(err, "failed to listen on %q", tmpAddr)
+		}
+
+		localRegistryAddr = fmt.Sprintf("localhost:%d", ln.Addr().(*net.TCPAddr).Port)
+
+		regProxy := &http.Server{
+			Handler: regproxy.NewRegistryProxy(bkclient.RegistryClient()),
+		}
+
+		a.cli.Console().VerbosePrintf("Starting local registry proxy: %s", localRegistryAddr)
+
+		go func() {
+			err := regProxy.Serve(ln)
+			if err != nil && !errors.Is(err, http.ErrServerClosed) {
+				a.cli.Console().Warnf("Failed to serve registry proxy: %v", err)
+			}
+		}()
+
+		defer func() {
+			ctx := context.Background()
+			err := regProxy.Shutdown(ctx)
+			if err != nil {
```

Comment:
Maybe don't warn if the err is a cancellation.

Reply:
Good call. Thanks.